needle.pm

```diff
@@ -239,19 +233,20 @@ sub init {
 }
 
 sub tags {
-    my @tags      = split(/ /, shift);
-    my $first_tag = shift @tags;
+    my ($wanted) = @_;
+    my @wanted    = split(/ /, $wanted);
+    my $first_tag = shift @wanted;
     my $goods     = $tags{$first_tag};
 
     # go out early if there is nothing to do
-    if (!$goods || !@tags) {
+    if (!$goods || !@wanted) {
         return $goods || [];
     }
     my @results;
 
     # now check that it contains all the other tags too
   NEEDLE: for my $n (@$goods) {
-        for my $t (@tags) {
+        for my $t (@wanted) {
```

Comment:
thank you for replacing the ambiguous `@tags` within the method `tags`. I got confused by it already :-)